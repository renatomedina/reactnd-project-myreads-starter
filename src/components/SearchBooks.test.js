import React from 'react';
import { shallow, mount } from 'enzyme';
// import SearchBooks from './SearchBooks';

describe('<SearchBooks />', () => {
    xit('shallow renders correctly', () => {
      expect(shallow(<SearchBooks />))
    })


})