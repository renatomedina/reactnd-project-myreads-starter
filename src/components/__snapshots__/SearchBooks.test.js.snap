// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBooks /> shallow renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBooks
    addBookHandler={[Function]}
    libaryBooks={
      Array [
        Object {
          "allowAnonLogging": true,
          "authors": Array [
            "William E. Shotts, Jr.",
          ],
          "averageRating": 4,
          "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
          "categories": Array [
            "COMPUTERS",
          ],
          "contentVersion": "1.2.2.0.preview.2",
          "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
          "id": "nggnmAEACAAJ",
          "imageLinks": Object {
            "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
            "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
          },
          "industryIdentifiers": Array [
            Object {
              "identifier": "9781593273897",
              "type": "ISBN_13",
            },
            Object {
              "identifier": "1593273894",
              "type": "ISBN_10",
            },
          ],
          "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
          "language": "en",
          "maturityRating": "NOT_MATURE",
          "pageCount": 480,
          "panelizationSummary": Object {
            "containsEpubBubbles": false,
            "containsImageBubbles": false,
          },
          "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
          "printType": "BOOK",
          "publishedDate": "2012",
          "publisher": "No Starch Press",
          "ratingsCount": 2,
          "readingModes": Object {
            "image": false,
            "text": true,
          },
          "shelf": "currentlyReading",
          "subtitle": "A Complete Introduction",
          "title": "The Linux Command Line",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-books-bar"
        >
          <Link
            className="close-search"
            replace={false}
            to="/"
          >
            Close
          </Link>
          <div
            className="search-books-input-wrapper"
          >
            <Debounce
              handler="onChange"
              handlers={Array []}
              time="400"
            >
              <input
                onChange={[Function]}
                placeholder="Search by title or author"
                type="text"
              />
            </Debounce>
          </div>
        </div>,
        <div
          className="search-books-results"
        >
          <ol
            className="books-grid"
          />
        </div>,
        <Unknown
          books={Array []}
          onChangeShelf={[Function]}
        />,
      ],
      "className": "search-books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="close-search"
              replace={false}
              to="/"
            >
              Close
            </Link>,
            <div
              className="search-books-input-wrapper"
            >
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  onChange={[Function]}
                  placeholder="Search by title or author"
                  type="text"
                />
              </Debounce>
            </div>,
          ],
          "className": "search-books-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "className": "close-search",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  onChange={[Function]}
                  placeholder="Search by title or author"
                  type="text"
                />
              </Debounce>,
              "className": "search-books-input-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <input
                  onChange={[Function]}
                  placeholder="Search by title or author"
                  type="text"
                />,
                "handler": "onChange",
                "handlers": Array [],
                "time": "400",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Search by title or author",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ol
            className="books-grid"
          />,
          "className": "search-books-results",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "books-grid",
          },
          "ref": null,
          "rendered": null,
          "type": "ol",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "books": Array [],
          "onChangeShelf": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-books-bar"
          >
            <Link
              className="close-search"
              replace={false}
              to="/"
            >
              Close
            </Link>
            <div
              className="search-books-input-wrapper"
            >
              <Debounce
                handler="onChange"
                handlers={Array []}
                time="400"
              >
                <input
                  onChange={[Function]}
                  placeholder="Search by title or author"
                  type="text"
                />
              </Debounce>
            </div>
          </div>,
          <div
            className="search-books-results"
          >
            <ol
              className="books-grid"
            />
          </div>,
          <Unknown
            books={Array []}
            onChangeShelf={[Function]}
          />,
        ],
        "className": "search-books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="close-search"
                replace={false}
                to="/"
              >
                Close
              </Link>,
              <div
                className="search-books-input-wrapper"
              >
                <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    onChange={[Function]}
                    placeholder="Search by title or author"
                    type="text"
                  />
                </Debounce>
              </div>,
            ],
            "className": "search-books-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Close",
                "className": "close-search",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="400"
                >
                  <input
                    onChange={[Function]}
                    placeholder="Search by title or author"
                    type="text"
                  />
                </Debounce>,
                "className": "search-books-input-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <input
                    onChange={[Function]}
                    placeholder="Search by title or author"
                    type="text"
                  />,
                  "handler": "onChange",
                  "handlers": Array [],
                  "time": "400",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onChange": [Function],
                    "placeholder": "Search by title or author",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ol
              className="books-grid"
            />,
            "className": "search-books-results",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "books-grid",
            },
            "ref": null,
            "rendered": null,
            "type": "ol",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "books": Array [],
            "onChangeShelf": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
